src/Microsoft.AspNetCore.Sockets.Abstractions/ConnectionContext.cs

```diff
@@ -1,25 +1,20 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
-using System.Security.Claims;
+using System.Collections.Generic;
 using System.Threading.Tasks.Channels;
 using Microsoft.AspNetCore.Http.Features;
 
 namespace Microsoft.AspNetCore.Sockets
 {
     public abstract class ConnectionContext
     {
-        public abstract string ConnectionId { get; }
+        public abstract string ConnectionId { get; set; }
```

Comment:
Like most (if not all) the things on the HttpContext, this is settable so that middleware can mutate.